### ISSUE TITLE
Future goals

### ISSUE THREAD

**Issue by klmurphy72** [Joylei/eip-rs#5]
I happened across this lib just about the time you made your first commit.  I am impressed by the rapid development.

I am wonder what your future plans/goals are for this lib.

I actually started writing some code to use your early work to read the template / structure data out of the processor.  I then took another look at the repository, and BAM......already there!!!!!  Impressive!!!!

**Comment by Joylei** [comment]
Great to hear that it worked for you.

The most recent goals:
- stabilize the features and Apis
- publish the crates to crates.io

The code currently are still under heavily refactoring. All features listed in the readme will be the features supported by this project. No new features will be added in the near feature. Only generic CIP and AB devices are officially supported by this project. Because I only have and work with AB PLCs. Other PLC devices support is out of scope for now.

As said above, the code must be stabilized, so some of the untested and unverified code will be removed.

Further considerations for this project:
- different futures runtime support, like smol
the project depends on tokio-util. To make it independent from tokio-util, it need to make a own copy of Framed/UdpFramed. I've looked at asynchronous-codec, its constrains are more strict than tokio-util, and it's impossible to move to it directly. so either make a own copy of Framed/UdpFramed or find a way make it work.
- no-std
- DF1 support
I have looked at it for a while. No decision made for now.
- IO connection support
- CIP Server side implementation
